### DISCUSSION TITLE
What is the purpose of optimizer.averages?

### DISCUSSION THREAD

**Discussion by echatzikyriakidis** [explosion/spaCy#6588] [Help: Coding & Implementations]
Can someone explain to me what is the purpose of:

```
with my_model.model.use_params(optimizer.averages)
```

I have seen it here: https://github.com/explosion/spaCy/blob/master/examples/training/train_textcat.py

Also read it here: https://spacy.io/usage/training#tips

But cannot understand exactly where and when we need to use it.

Also, I want to train a NER model and the following training/updating NER examples do not use `optimizer.averages`:

- https://github.com/explosion/spaCy/blob/master/examples/training/train_ner.py
- https://github.com/explosion/spaCy/blob/master/examples/training/train_new_entity_type.py

**Comment by svlandeg** [comment]
FYI - I converted this question to a thread on the discussion board. 

We're aiming to keep the Issue tracker focused mainly on bug reports, while the new GH Discussion board can be a place where the community can come together, help eachother, discuss best practices, and so on.

**Reply by echatzikyriakidis** [reply]
@svlandeg 

Could you give me some insight on this?

**Reply by svlandeg** [reply]
Could you please avoid pinging maintainers? It makes our GitHub notifications unusable. Also, please consider that this thread is merely 6 days old. While we spend a lot of effort in maintaining the issue tracker and the discussions forum, we ofcourse have other work to do as well, and I'm sure there are other people who can help with this.
